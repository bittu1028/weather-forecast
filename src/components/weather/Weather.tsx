import React from 'react';
import { City, CurrentWeather } from '../../models/weather.model';
import { H2, H4, StyledHeader, StyledLocation, StyledTemperatureInfoContainer, StyledWeatherContainer, StyledWeatherWrapper, TemperatureContainer } from './StyledWeather';

 
interface WeatherProp {
  onToggle: () => void;
  weather: CurrentWeather;
  isCelcius?: boolean;
  cityInfo?: City;
}

const Weather = ({
  onToggle, 
  weather, 
  isCelcius = true, 
  cityInfo
}: WeatherProp)  => {
  const {
    currentTemp,
    weatherInfo,
    time,
    wind,
  } = weather;

  const countryFlagsUrl = 'https://www.countryflags.io/';

  return (
    <StyledWeatherContainer>
      <StyledWeatherWrapper>
        {/* <i className={`wi ${weatherbitIcons[weatherIconCode]}`} /> */}
        <TemperatureContainer>
          <StyledHeader>
            {isCelcius ? `${currentTemp?.temp} 째C` : `${currentTemp?.temp} 째F`} 
          </StyledHeader>
            <div 
                className="temperature-toggle" 
                onClick={onToggle}
                style={{
                  color: isCelcius ? 'rgba(255, 255, 255, .7)' : '#adff2f'
                }}
            >
              <span>{isCelcius ? '째F' : '째C' }</span>
            </div>
        </TemperatureContainer>
      </StyledWeatherWrapper>
      <StyledTemperatureInfoContainer>
        <StyledLocation>
          <H2>{cityInfo?.name}, {cityInfo?.country}</H2>
          <img src={`${countryFlagsUrl}/${cityInfo?.country}/shiny/64.png`} alt=""/>
        </StyledLocation>
        <H4>
          <span>Timezone:</span> {cityInfo?.country}
        </H4>
        <H4>
          <span>Weather:</span> {weatherInfo}
        </H4>
        <H4><span>Wind Speed:</span> {wind.speed} km/h</H4>
        <H4><span>Humidity:</span> {currentTemp?.humidity}%</H4>
        <H4><span>Date:</span> {time}</H4>
      </StyledTemperatureInfoContainer>
    </StyledWeatherContainer>
  );
}

export default Weather;