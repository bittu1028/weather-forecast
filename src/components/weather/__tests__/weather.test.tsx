// __tests__/searchBox.test.js
import React from 'react';
import {
  render, screen,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Weather from '../Weather';
import { currentWeatherData, mockCityData } from '../../../test/mockData';

const onToggleFahrenheit = jest.fn();


describe('Current Weather', () => {
    test('renders current temperature component', async () => {
        const { asFragment } = render(<Weather  
            onToggle={onToggleFahrenheit}
            weather={currentWeatherData}
            cityInfo={mockCityData}
        />);
        expect(asFragment).toMatchSnapshot();
    });

  test('City name and country code should be present', async () => {
    render(<Weather  
        onToggle={onToggleFahrenheit}
        weather={currentWeatherData}
        cityInfo={mockCityData}
    />);
    const text = screen.getByText('Sydney, AU');
    expect(text).toBeInTheDocument();
  });

  // test('Expect forecast to render for 5 days', async () => {
  //   render(<Weather  
  //       onToggle={onToggleFahrenheit}
  //       weather={currentWeatherData}
  //       cityInfo={mockCityData}
  //   />);
  //   const list = screen.getAllByRole('card-list-item');
  //   console.log(list);
  //   expect(list.length).toEqual(4);
  // });
  
});
