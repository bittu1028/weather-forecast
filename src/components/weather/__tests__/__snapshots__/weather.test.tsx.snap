// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Weather renders current temperature component 1`] = `[Function]`;
